tests/is/L42/perftests/PerfTests.java

```diff
@@ -0,0 +1,167 @@
+      } 
+    catch (Exception e) {
+      throw new Error(e);
+      }
+    }
+  
+  @AfterAll
+  public static void showLog() {
+    System.setOut(out);
+    System.setErr(err);
+    System.out.println("L42 Program output logged to: " + programOutput.getAbsolutePath());
+    }
+  
+  @Test 
+  public void testAdamsTowel() throws IOException, URISyntaxException, ClassNotFoundException {
+    testRun("L42Source/AdamsTowel", "adamstowel");
+    }
+
+  @Test 
+  public void testUnit() throws IOException, URISyntaxException, ClassNotFoundException {
+    testRun("L42Source/TestUnit", "unit");
+    }
+  
+  @Test 
+  public void testLoad() throws IOException, URISyntaxException, ClassNotFoundException {
+    testRun("L42Source/TestLoad", "load");
+    }
+  
+  @Test 
+  public void testHelloWorld() throws IOException, URISyntaxException, ClassNotFoundException {
+    testRun("L42Source/HelloWorld2", "helloworld");
+    }
+  
+  @Test 
+  public void testCacheSaving() throws IOException, URISyntaxException, ClassNotFoundException
+  {
+    PerfStat prevres = PerfStat.getResult("cachesaveload");
+    PerfStat prevres2 = PerfStat.getResult("unit.memcache");
+    PerfStat prevres3 = PerfStat.getResult("unit.filecache");
+    File tmp = new File("tmp/tmp" + Math.random()); tmp.mkdirs();
+    var url=Paths.get(DeployAdamsTowel.class.getResource("L42Source/TestUnit").toURI()).resolve("This.L42");
+    var code=Parse.codeFromPath(url);
+    CachedTop memcache = new CachedTop(L(),L());
+    Resources.clearRes();
+    Init.topCache(memcache, url, code);
+    Resources.clearResKeepReuse();
+    final CachedTop finalvar = memcache;
+    PerfResult res = PerfMonitor.test(() -> {
+      finalvar.toNextCache().saveCache(tmp.toPath());
+      return CachedTop.loadCache(tmp.toPath());
+      }, 5, true);
+    CachedTop filecache = res.cache();
+    PerfStat res2 = PerfMonitor.test(() -> {
+      doExecCache(finalvar.toNextCache(), url, code);
+      return null;
+      }, 5, true).stat();
+    PerfStat res3 = PerfMonitor.test(() -> {
+      doExecCache(filecache, url, code);
+      return null;
+      }, 1, true).stat();
+    System.out.println("Cache saving and loading: " + res.stat().toString());
+    System.out.println("Unit with in-memory cache: " + res2.toString());
+    System.out.println("Unit with file cache: " + res3.toString());
+    if(SAVE_RESULTS) {
+      PerfStat.saveResult("cachesaveload", res.stat());
+      PerfStat.saveResult("unit.memcache", res2);
+      PerfStat.saveResult("unit.filecache", res3);
+      } else {
+      if(prevres != null) {
+        assertTrue(prevres.compareTo(res.stat()) <= 0);
+        }
+      if(prevres2 != null) {
+        assertTrue(prevres.compareTo(res2) <= 0);
+        }
+      if(prevres3 != null) {
+        assertTrue(prevres.compareTo(res3) <= 0);
+        }
+      }
+    }
+  
+  public void testRun(String path, String name) throws IOException, URISyntaxException, ClassNotFoundException {
```

Comment:
do you expect those exceptions to actually raise? I think not, so you can try-catch them error.